embulk-input-jdbc/src/main/java/org/embulk/input/jdbc/AbstractJdbcInputPlugin.java

```diff
@@ -362,6 +367,16 @@ private boolean fetch(BatchSelect cursor,
                 reportRows *= 2;
             }
         } while (result.next());
+
+        if (task.getIncrementalColumn().isPresent()) {
+            for (String key : task.getIncrementalColumn().get()) {
+                if (temporaryLastRecord.containsKey(key)) {
```

Comment:
This seems invalid because column names are case insensitive in SQL. If column names of the target table don't match with selecting column names, column names in the results may not match with selecting column names.